File: timeplan/front/src/app/models/teacher.ts
import {Subject} from "./subject";

export class Teacher{
  idteacher:number;
  fname:string;
  lname:string;
  email:string;
  password:string;
  level:string;
  subjects: Subject[]=[];

  constructor(){}


  static cloneStudent(teacher: Teacher):Teacher{
    let copyTeacher = new Teacher();
   copyTeacher.idteacher = teacher.idteacher;
   copyTeacher.fname = teacher.fname;
   copyTeacher.lname = teacher.lname;
   copyTeacher.level = teacher.level;
   copyTeacher.email = teacher.email;
   copyTeacher.password =teacher.password;
   copyTeacher.subjects = teacher.subjects;
   return copyTeacher;
  }

  getInfo():string{
    return this.fname + " " + this.lname
  }
}

File: timeplan/fapi/src/main/java/com/netcracker/fapi/model/Teacher.java
package com.netcracker.fapi.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import java.util.List;

@JsonIgnoreProperties(ignoreUnknown = true)
public class Teacher {
    private Long idteacher;
    private String fname;
    private String lname;
    private String email;
    private String password;
    private String level;
    private String role;
    private List<Subject> subjects;


    public Teacher() {
    }

    public String getRole() {
        return role;
    }

    public void setRole(String role) {
        this.role = role;
    }

    public Long getIdteacher() {
        return idteacher;
    }

    public void setIdteacher(Long idteacher) {
        this.idteacher = idteacher;
    }

    public String getFname() {
        return fname;
    }

    public void setFname(String fname) {
        this.fname = fname;
    }

    public String getLname() {
        return lname;
    }

    public void setLname(String lname) {
        this.lname = lname;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public String getLevel() {
        return level;
    }

    public void setLevel(String level) {
        this.level = level;
    }

    public List<Subject> getSubjects() {
        return subjects;
    }

    public void setSubjects(List<Subject> subjects) {
        this.subjects = subjects;
    }
}
